File: descend_calc/descend_calc.go
package main

import "fmt"

func get_input(msg string) int {
	var input int

	for true {
		fmt.Print(msg)

		_, err := fmt.Scanln(&input)
		if err != nil {
			var discard string
			fmt.Scanln(&discard)
		} else {
			break
		}
	}

	return input
}

func main() {
	var current_altitude = get_input("Current altitude: ")
    var airport_altitude = get_input("Airport altitude: ")

	var altitude_to_lose = (float32(current_altitude - airport_altitude) / 100) / 3;

	var current_speed = get_input("Current speed: ");
    var avg_dsc_speed = get_input("Average descend speed: ");

	var speed_knots = float32(current_speed - avg_dsc_speed) / 10

	var wind_speed = float32(get_input("Wind speed: ")) / 10
    var tail_wind = get_input("Tail or head wind [1 for tail | 0 for head]: ")
    
	var descend_distance = altitude_to_lose + speed_knots
	if tail_wind == 1 {
		descend_distance += wind_speed
	} else {
		descend_distance -= wind_speed
	}

	fmt.Println("------------------------------")
	fmt.Printf("Altitude to lose nm: %v\n", altitude_to_lose)
	fmt.Printf("Speed nm: %v\n", speed_knots)
	fmt.Printf("Wind  nm: %v\n", wind_speed)
	fmt.Printf("Total distance nm: %v\n", descend_distance)
}

File: descend_calc/descend_calc.rs
use std::io;
use std::io::Write;

fn get_input(msg: &str) -> i32
{
    loop
    {
        let mut input = String::new();

        print!("{}", msg);
        io::stdout().flush().unwrap();

        let _ = io::stdin().read_line(&mut input);
        match input.trim().parse::<i32>() {
            Ok(n) => return n,
            Err(_e) => continue,
        }
    }
}

fn main()
{
    let current_altitude = get_input("Current altitude: ");
    let airport_altitude = get_input("Airport altitude: ");

    let altitude_to_lose: f32 = ((current_altitude - airport_altitude) as f32 / 100.0f32) / 3.0f32;

    let current_speed = get_input("Current speed: ");
    let avg_dsc_speed = get_input("Average descend speed: ");

    let speed_knots: f32 = (current_speed - avg_dsc_speed) as f32 / 10.0f32;

    let wind_speed: f32 = get_input("Wind speed: ") as f32 / 10.0f32;
    let tail_wind = get_input("Tail or head wind [1 for tail | 0 for head]: ");
    
    let descend_distance: f32 = altitude_to_lose + speed_knots + (if tail_wind == 1 { wind_speed } else { -wind_speed });

    println!("------------------------------");
    println!("Altitude to lose nm: {}", altitude_to_lose);
    println!("Speed nm: {}", speed_knots);
    println!("Wind  nm: {}", wind_speed);
    println!("Total distance: {}", descend_distance);
}